# Copyright (c) 2006, 2009, Oracle and/or its affiliates. All rights reserved.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the "Software"),
# to deal in the Software without restriction, including without limitation
# the rights to use, copy, modify, merge, publish, distribute, sublicense,
# and/or sell copies of the Software, and to permit persons to whom the
# Software is furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice (including the next
# paragraph) shall be included in all copies or substantial portions of the
# Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
# DEALINGS IN THE SOFTWARE.
Redo the patch for Xorg 1.5. The patch is helpful only for nv driver.

diff -Nurp -x '*~' -x '*.orig' hw/xfree86/modes/xf86EdidModes.c hw/xfree86/modes/xf86EdidModes.c
diff -urp -x '*~' -x '*.orig' hw/xfree86/modes/xf86EdidModes.c hw/xfree86/modes/xf86EdidModes.c
--- hw/xfree86/modes/xf86EdidModes.c	2009-11-26 21:40:35.000000000 -0800
+++ hw/xfree86/modes/xf86EdidModes.c	2009-12-02 18:58:15.024979778 -0800
@@ -948,6 +948,20 @@ xf86DDCGetModes(int scrnIndex, xf86MonPt
                                              &det_mon->section.d_timings,
 					     preferred,
 					     quirks);
+
+	    if (Mode && (DDC->features.input_type == 0) &&
+            	(((Mode->HDisplay * 5 / 4) & ~0x07) > Mode->HTotal)) {
+		struct detailed_timings *dt = &DDC->det_mon[i].section.d_timings;
+		DisplayModeRec *cvtmode;
+
+		cvtmode = xf86CVTMode(dt->h_active, dt->v_active,
+                                (double) xf86ModeVRefresh(Mode), TRUE, FALSE);
+		memcpy(Mode, cvtmode, sizeof(DisplayModeRec));
+		Mode->type       = M_T_DRIVER;
+    		if (preferred)
+		    Mode->type |= M_T_PREFERRED;
+	    }
+		
 	    preferred = FALSE;
             break;
         case DS_STD_TIMINGS:
